bcipy/simulator/ui/gui.py

```diff
@@ -0,0 +1,679 @@
+            child.widget().deleteLater()
+        elif child.layout() is not None:
+            clear_layout(child.layout())
+
+
+class MainPanel(QWidget):
+    """Main GUI window.
+    Parameters:
+    -----------
+      json_file: path to the parameters file.
+      title: window title
+      size: window size; (width, height)
+    """
+
+    def __init__(self, title: str, size: Tuple[int, int]):
+        super().__init__()
+        self.title = title
+        self.size = size
+
+        self.command = ''
+
+        self.form = SimConfigForm(width=self.size[0],
+                                  change_event=self.on_params_change)
+        self.flash_msg = ''
+        self.init_ui()
+
+    def init_ui(self):
+        """Initialize the UI"""
+        vbox = QVBoxLayout()
+
+        header_box = QHBoxLayout()
+        header_box.addSpacing(4)
+        header_box.addWidget(
+            static_text_control(self,
+                                label='BciPy Simulator Configuration',
+                                size=16,
+                                color='dimgray'))
+        vbox.addLayout(header_box)
+        vbox.addWidget(self.form)
+        vbox.addSpacing(4)
+
+        # Controls
+        control_box = QHBoxLayout()
+        control_box.addStretch()
+        self.run_button = QPushButton('Run')
+        self.run_button.setFixedWidth(80)
+        self.run_button.clicked.connect(self.on_run)
+        self.run_button.setEnabled(False)
+        control_box.addWidget(self.run_button)
+
+        vbox.addLayout(control_box)
+
+        self.command_msg = static_text_control(self,
+                                               label='',
+                                               size=12,
+                                               color='darkgray')
+        vbox.addWidget(self.command_msg)
+        self.setLayout(vbox)
+        self.setFixedHeight(self.size[1])
+        self.setWindowTitle(self.title)
+        self.show()
+
+    def cli_output(self) -> str:
+        """Returns the command with arguments."""
+        if self.form.command_valid():
+            return self.form.command()
+        return ''
+
+    def on_run(self):
+        """Event handler for running the simulation."""
+
+        if self.form.command_valid():
+            self.command_msg.setText(self.cli_output())
+            self.command_msg.repaint()
+            self.close()
+
+    def on_params_change(self) -> None:
+        """Event performed when form parameters change."""
+        if self.form.command_valid():
+            self.run_button.setEnabled(True)
+        else:
+            self.run_button.setEnabled(False)
+
+
+def init(title='BCI Simulator', size=(750, 600)) -> str:
+    """Set up the GUI components and start the main loop."""
+    app = QApplication(sys.argv)
+    panel = MainPanel(title, size)
+    app.exec()
+    command = panel.cli_output()
+    app.quit()
+    if command:
+        print(command, file=sys.stdout)
```

Comment:
use the logger?

Reply:
Since this starts executing a subprocess immediately in the console it's confusing to the user if this doesn't print to stdout. Also, this gives users the opportunity to run straight from the command line once the configuration is set, which can be faster than the GUI if you're doing this multiple times with only small variation.